js/adapt-contrib-xapi.js

```diff
@@ -64,1568 +61,1516 @@ define([
-
-      return url;
-    },
-
-    getAttemptDuration: function() {
-      return this.startAttemptDuration + this.getSessionDuration();
-    },
-
-    getSessionDuration: function() {
-      return Math.abs((new Date()) - this.startTimeStamp);
-    },
-
-    /**
-     * Converts milliseconds to an ISO8601 duration
-     * @param {int} inputMilliseconds - Duration in milliseconds
-     * @return {string} - Duration in ISO8601 format
-     */
-    convertMillisecondsToISO8601Duration: function(inputMilliseconds) {
-      var hours;
-      var minutes;
-      var seconds;
-      var i_inputMilliseconds = parseInt(inputMilliseconds, 10);
-      var i_inputCentiseconds;
-      var inputIsNegative = '';
-      var rtnStr = '';
-
-      // Round to nearest 0.01 seconds.
-      i_inputCentiseconds = Math.round(i_inputMilliseconds / 10);
-
-      if (i_inputCentiseconds < 0) {
-        inputIsNegative = '-';
-        i_inputCentiseconds = i_inputCentiseconds * -1;
-      }
+    });
 
-      hours = parseInt(((i_inputCentiseconds) / 360000), 10);
-      minutes = parseInt((((i_inputCentiseconds) % 360000) / 6000), 10);
-      seconds = (((i_inputCentiseconds) % 360000) % 6000) / 100;
+    if (!newConfig.length > 0) {
+      this.xapiWrapper.changeConfig(newConfig);
 
-      rtnStr = inputIsNegative + 'PT';
-      if (hours > 0) {
-        rtnStr += hours + 'H';
+      if (!this.xapiWrapper.testConfig()) {
+        throw new Error('Incorrect xAPI configuration detected');
       }
+    }
+  }
 
-      if (minutes > 0) {
-        rtnStr += minutes + 'M';
-      }
+  /**
+   * Gets the URL the course is currently running on.
+   * @return {string} The URL to the current course.
+   */
+  getBaseUrl() {
+    const url = window.location.origin + window.location.pathname;
 
-      rtnStr += seconds + 'S';
+    Adapt.log.info(`adapt-contrib-xapi: Using detected URL (${url}) as ActivityID`);
 
-      return rtnStr;
-    },
+    return url;
+  }
 
-    setupListeners: function() {
-      if (!this.get('isInitialised')) {
-        Adapt.log.warn('adapt-contrib-xapi: Unable to setup listeners for xAPI');
-        return;
-      }
+  getAttemptDuration() {
+    return this.startAttemptDuration + this.getSessionDuration();
+  }
+  
+  getSessionDuration() {
+    return Math.abs((new Date()) - this.startTimeStamp);
+  }
 
-      // Allow surfacing the learner's info in _globals.
-      this.getLearnerInfo()
+  /**
+   * Converts milliseconds to an ISO8601 duration
+   * @param {int} inputMilliseconds - Duration in milliseconds
+   * @return {string} - Duration in ISO8601 format
+   */
+  convertMillisecondsToISO8601Duration(inputMilliseconds) {
+    let hours;
+    let minutes;
+    let seconds;
+    const i_inputMilliseconds = parseInt(inputMilliseconds, 10);
+    let i_inputCentiseconds;
+    let inputIsNegative = '';
+    let rtnStr = '';
+
+    // Round to nearest 0.01 seconds.
+    i_inputCentiseconds = Math.round(i_inputMilliseconds / 10);
```

Comment:
Minor: The let i_inputCentiseconds; can be removed from above, and this can be the first declaration of the variable